auto3dseg/algorithm_templates/dints/scripts/train.py

```diff
@@ -116,14 +116,146 @@ def __call__(self, val_acc):
+
+    def __call__(self, val_acc):
+        if self.best_score is None:
+            self.best_score = val_acc
+        elif val_acc + self.delta < self.best_score:
+            self.counter += 1
+            if self.verbose:
+                logger.debug(
+                    f"EarlyStopping counter: {self.counter} out of {self.patience}"
+                )
+            if self.counter >= self.patience:
+                self.early_stop = True
+        else:
+            self.best_score = val_acc
+            self.counter = 0
+
+
+def get_mem_from_visible_gpus():
+    available_mem_visible_gpus = []
+    for d in range(torch.cuda.device_count()):
+        available_mem_visible_gpus.append(torch.cuda.mem_get_info(device=d)[0])
+    return available_mem_visible_gpus
+
+
+def pre_operation(config_file, **override):
+    # update hyper-parameter configuration
+    rank = int(os.getenv("RANK", "0"))
+    if rank == 0:
+        if isinstance(config_file, str) and ',' in config_file:
+            config_file = config_file.split(',')
+
+        for _file in config_file:
+            if "hyper_parameters.yaml" in _file:
+                parser = ConfigParser(globals=False)
+                parser.read_config(_file)
+
+                auto_scale_allowed = parser["training"]["auto_scale_allowed"]
+                if "training#auto_scale_allowed" in override:
+                    auto_scale_allowed = override["training#auto_scale_allowed"]
+
+                if auto_scale_allowed:
+                    output_classes = parser["training"]["output_classes"]
+                    mem = get_mem_from_visible_gpus()
+                    mem = min(mem) if isinstance(mem, list) else mem
+                    mem = float(mem) / (1024.0**3)
+                    mem = max(1.0, mem - 1.0)
+                    mem_bs2 = 6.0 + (20.0 - 6.0) * \
+                        (output_classes - 2) / (105 - 2)
+                    mem_bs9 = 24.0 + (74.0 - 24.0) * \
+                        (output_classes - 2) / (105 - 2)
+                    batch_size = 2 + (9 - 2) * \
+                        (mem - mem_bs2) / (mem_bs9 - mem_bs2)
+                    batch_size = int(batch_size)
+                    batch_size = max(batch_size, 1)
+
+                    parser["training"].update(
+                        {"num_patches_per_iter": batch_size})
+                    parser["training"].update(
+                        {"num_patches_per_image": 2 * batch_size})
+                    parser["training"].update(
+                        {"num_epochs": int(400.0 / float(batch_size))})
+
+                    ConfigParser.export_config_file(
+                        parser.get(), _file, fmt="yaml", default_flow_style=None)
+
+    return
+
+
 def run(config_file: Optional[Union[str, Sequence[str]]] = None, **override):
+
     logging.basicConfig(stream=sys.stdout, level=logging.INFO)
 
+    # pre-operations
+    logger.debug(f"[info] number of GPUs: {torch.cuda.device_count()}")
```

Comment:
Please remove `[info]` from the logging.debug

Reply:
I have removed all similar instances from the respective lines.